src/core/impl.scala

```diff
@@ -24,47 +24,66 @@ object CaseClassDerivation:
         defaultValue[A].to(Map)
       )*
     )
-
-    new CaseClass(
+    ProductCaseClass(
```

Comment:
sounds awesome! Out of curiosity, can you share some insight on *why* this improves performance? maybe even as a comment, for informing future reads and avoiding introducing regressions

I'm guessing that we're not generating a new anonymous class construction every time, but instead we generate an invocation of the constructor, right? but why does this have such a performance impact?

Reply:
i think it's just the amount of inlined code, compiler has to compile it for every method invocation
but i'm not really sure...

Reply:
Well you need a separate classfile for every anonymous instance so it can add up.

Reply:
thanks, I guess the simplest explanation is the the right one ;)